### PR TITLE
Package box.1.2.0

### PR DESCRIPTION
**Pull request by davesnx**
### `box.1.2.0`
Render boxes in the terminal
Render boxes in the terminal, can configure the padding, margin and the border of the box.

---
* Homepage: https://github.com/davesnx/ocaml-box
* Source repo: git+https://github.com/davesnx/ocaml-box.git
* Bug tracker: https://github.com/davesnx/ocaml-box/issues

---
:camel: Pull-request generated by opam-publish v2.1.0

### CODE REVIEW

File: packages/box/box.1.2.0/opam [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Render boxes in the terminal, can configure the padding, margin and the border of the box.
+"""
+maintainer: ["davesnx"]
+authors: ["davesnx"]
+license: "MIT"
+homepage: "https://github.com/davesnx/ocaml-box"
+bug-reports: "https://github.com/davesnx/ocaml-box/issues"
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.7" & >= "2.0"}
```
Comment (kit-ty-kate) [inline packages/box/box.1.2.0/opam:14]:
```suggestion
  "dune" {>= "2.7"}
```

File: packages/box/box.1.2.0/opam [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  "odoc" {with-doc}
+  "reason" {build}
+  "fmt" {dev}
+]
+dev-repo: "git+https://github.com/davesnx/ocaml-box.git"
+# We need to avoid "@runtest", since it depends on rely
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (kit-ty-kate) [inline packages/box/box.1.2.0/opam:22]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/box/box.1.2.0/opam [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+bug-reports: "https://github.com/davesnx/ocaml-box/issues"
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.7" & >= "2.0"}
+  "odoc" {with-doc}
+  "reason" {build}
+  "fmt" {dev}
+]
+dev-repo: "git+https://github.com/davesnx/ocaml-box.git"
+# We need to avoid "@runtest", since it depends on rely
```
Comment (kit-ty-kate) [inline packages/box/box.1.2.0/opam:20]: You could vendor `rely` if that's the only dependency that's not in opam

File: packages/box/box.1.2.0/opam [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+authors: ["davesnx"]
+license: "MIT"
+homepage: "https://github.com/davesnx/ocaml-box"
+bug-reports: "https://github.com/davesnx/ocaml-box/issues"
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.7" & >= "2.0"}
+  "odoc" {with-doc}
+  "reason" {build}
+  "fmt" {dev}
```
Comment (kit-ty-kate) [inline packages/box/box.1.2.0/opam:17]: I'm not seeing any use of `fmt` in your project. Why was that added?
```suggestion
```

File: packages/box/box.1.2.0/opam [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Render boxes in the terminal"
+description: """
+Render boxes in the terminal, can configure the padding, margin and the border of the box.
+"""
+maintainer: ["davesnx"]
```
Comment (kit-ty-kate) [inline packages/box/box.1.2.0/opam:7]: Any chance you could add some sort of email address for people who don't use github?

File: packages/box/box.1.2.0/opam (hunk 2 above)
Comment (davesnx) [reply packages/box/box.1.2.0/opam:22]: Yes https://github.com/davesnx/ocaml-box/blob/main/dune-project#L1

File: packages/box/box.1.2.0/opam (hunk 3 above)
Comment (davesnx) [reply packages/box/box.1.2.0/opam:20]: It's a test dependency, does it make sense to vendor it in this case?

Comment (kit-ty-kate) [reply packages/box/box.1.2.0/opam:20]: how so? A vendored package with dune won't be built unless it is needed (only when building the tests in this case)

File: packages/box/box.1.2.0/opam [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+authors: ["davesnx <dsnxmoreno@gmail.com>"]
+license: "MIT"
+homepage: "https://github.com/davesnx/ocaml-box"
+doc: "https://davesnx.github.io/ocaml-box/box/index.html"
+bug-reports: "https://github.com/davesnx/ocaml-box/issues"
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.7" & >= "2.7"}
```
Comment (kit-ty-kate) [inline packages/box/box.1.2.0/opam:15]: To remove this you need to remove the dependency on dune in the dune-project file
```suggestion
  "dune" {>= "2.7"}
```

File: packages/box/box.1.2.0/opam (hunk 5 above)
Comment (kit-ty-kate) [reply packages/box/box.1.2.0/opam:7]:
```suggestion
maintainer: ["davesnx <dsnxmoreno@gmail.com>"]
```

File: packages/box/box.1.2.0/opam (hunk 6 above)
Comment (kit-ty-kate) [reply packages/box/box.1.2.0/opam:15]: Alternatively you should not need the opam file autogeneration from dune so you can turn it off and modify the opam file directly instead.

File: packages/box/box.1.2.0/opam (hunk 3 above)
Comment (davesnx) [reply packages/box/box.1.2.0/opam]: I'm confused then, why did you vendor rely in ocaml-box then? Is it preferred that any consumable of a package can build the tests as well?

**Comment by mseri** [comment]
Looks good now, thanks

File: packages/box/box.1.2.0/opam (hunk 3 above)
Comment (kit-ty-kate) [reply packages/box/box.1.2.0/opam]: I’m unsure what you mean by "consumable of a package" but for opam-repository it is better when packages have tests, especially for libraries as this also detects if all the dependencies linked correctly.

My PR got merged (thanks! ^^) so it’ll be done for next release i suppose.
